### PR TITLE
UT to demonstrate caching data service fix

### PR DESCRIPTION
**Pull request by dsnyder-fairway**

### CODE REVIEW

File: webapp/src/main/webapp/src/app/shared/data/caching-data.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      expect(value).toEqual({ url: passedUrl, params: options.params });
+    });
+    service.get(passedUrl, options).subscribe((value) => {
+      expect(dataService.get).toHaveBeenCalledWith(passedUrl, options);
+      expect(dataService.get).toHaveBeenCalledTimes(1);
+      expect(value).toEqual({ url: passedUrl, params: options.params });
+    });
+  });
+
+  it('should handle different options', () => {
```
Comment (dsnyder-fairway) [inline webapp/src/main/webapp/src/app/shared/data/caching-data.service.spec.ts]: this test fails without my changes from #1150

File: webapp/src/main/webapp/src/app/shared/data/caching-data.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      expect(url).toBe(passedUrl);
+      return of({
+        url: passedUrl,
+        params: options.params
+      });
+    });
+
+    service.get(passedUrl, options1).subscribe((value) => {
+      expect(dataService.get).toHaveBeenCalledWith(passedUrl, options1);
+      expect(dataService.get).toHaveBeenCalled();
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/shared/data/caching-data.service.spec.ts]: i think `toHaveBeenCalled()` is redundant here with `toHaveBeenCalledWith()`

File: webapp/src/main/webapp/src/app/shared/data/caching-data.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+  it('should handle options', () => {
+    let passedUrl: string = "/something";
+    let options = {
+      params: {
+        oneParam: "oneValue"
+      }
+    };
+    dataService.get.and.callFake((url, options) => {
+      expect(url).toBe(passedUrl);
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/shared/data/caching-data.service.spec.ts]: not sure we should assert in the stubbed method

File: webapp/src/main/webapp/src/app/shared/data/caching-data.service.spec.ts (hunk 2 above)
Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/shared/data/caching-data.service.spec.ts]: 👍 I simplified this to reflect what the intention of each test is. If followed by `toHaveBeenCalledTimes(1)` is to show we hit the cache. When using  `toHaveBeenCalledWith` demonstrates we are populating the cache

File: webapp/src/main/webapp/src/app/shared/data/caching-data.service.spec.ts (hunk 3 above)
Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/shared/data/caching-data.service.spec.ts]: removed

**Pull request merged**